File: client/src/main/java/commands/pcommands/Help.java
package commands.pcommands;

import commands.*;

import java.util.stream.Collectors;

public final class Help extends AbstractCommand {
  private final CommandManager commandManager;

  public Help(CommandManager commandManager) {
    super(CommandInfo.of("help", "Shows info about accessible commands", true, 0, CommandType.OTHER, ExecutionMode.CLIENT));
    this.commandManager = commandManager;
  }

  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    CommandNameToInfo nameToCommand = commandManager.getUseraccessibleCommandsInfo();
    int longestNameLength = nameToCommand.keySet().stream().map(String::length).max(Integer::compareTo).orElse(27);
    String commandsWithDescriptions = nameToCommand.values().stream()
        .map(c -> String.format("%" + longestNameLength + "s: %s",
            c.getName(), c.getDescription()))
        .collect(Collectors.joining(System.lineSeparator()));
    return ExecutionResult.valueOf(true, commandsWithDescriptions);
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/Help.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*

class Help(private val commandManager: CommandManager) : AbstractCommand(
    CommandInfo.of(
        "help",
        "Shows info about accessible commands",
        true,
        0,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    override fun execute(payload: ExecutionPayload): ExecutionResult {
        val nameToCommand = commandManager.userAccessibleCommandsInfo
        val longestNameLength: Int = nameToCommand.maxOf { it.value.name.length }
        val commandsWithDescriptions: String = nameToCommand.values.map {
           info ->
            String.format("%" + longestNameLength + "s: %s", info.name, info.description)
        }.joinToString(separator = System.lineSeparator())
        return ExecutionResult.valueOf(true, commandsWithDescriptions)
    }
}